skops/card/_model_card.py

```diff
@@ -1,61 +1,189 @@
-import os
 import re
+import shutil
+import tempfile
+from pathlib import Path
 
-from modelcards import ModelCard
+from modelcards import CardData, ModelCard
 from sklearn.utils import estimator_html_repr
 
 import skops
 
 
-def _extract_estimator_config(model):
-    """Extracts estimator configuration and renders them into a vertical table.
-
-    Parameters
+class Card:
+    """Model card class that will be used to generate model card.
+    This class can be used to write information and plots to model card and save
+    it. This class by default generates an interactive plot of the model and a
+    table of hyperparameters. The slots to be filled are defined in the markdown
+    template.
+    Parameters:
     ----------
-        model (estimator): scikit-learn pipeline or model.
+    model: estimator object
+        Model that will be documented.
+    model_diagram: bool, optional
+        Set to True if model diagram should be plotted in the card.
+    Notes
+    -----
+    You can pass your own custom template using ``add`` method. You can add
+    plots to the model card template using ``add_plot``. The key you pass to
+    ``add_plot`` will be used for header of the plot.
 
-    Returns
-    -------
-    str:
-        Markdown table of hyperparameters.
+    Examples
+    --------
+    >>> from sklearn.metrics import ConfusionMatrixDisplay, confusion_matrix
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> import matplotlib.pyplot as plt
+    >>> from skops import card
+    >>> X, y = load_iris(return_X_y=True)
+    >>> model = LogisticRegression(random_state=0).fit(X, y)
+    >>> model_card = card.Card(model)
+    >>> model_card.add(license="mit") # doctest: +ELLIPSIS
+    <skops.card._model_card.Card object at ...>
+    >>> y_pred = model.predict(X)
+    >>> cm = confusion_matrix(y, y_pred,labels=model.classes_)
+    >>> disp = ConfusionMatrixDisplay(confusion_matrix=cm,
+    ... display_labels=model.classes_)
+    >>> disp.plot() # doctest: +ELLIPSIS
+    <sklearn.metrics._plot.confusion_matrix.ConfusionMatrixDisplay object at ...>
+    >>> plt.savefig("confusion_matrix.png") # doctest: +ELLIPSIS
+    ...
+    >>> model_card.add_plot(confusion_matrix="confusion_matrix.png") # doctest: +ELLIPSIS
+    <skops.card._model_card.Card object at ...>
+    >>> model_card.save((Path("save_dir") / "README.md")) # doctest: +ELLIPSIS
+    ...
```

Comment:
We can enable ELLIPSIS everywhere in our config file. Opened https://github.com/skops-dev/skops/issues/58 to track this. Nothing to do here.